src/box/memtx_tx.c

```diff
@@ -2037,13 +2047,15 @@ memtx_tx_history_add_insert_stmt(struct txn_stmt *stmt,
 	for (uint32_t i = 0; i < space->index_count; i++) {
 		struct tuple *next = directly_replaced[i];
 		struct tuple *succ = direct_successor[i];
-		if (next == NULL) {
-			/* Collect conflicts. */
+		/* Collect conflicts. */
+		if (next == NULL && !tuple_excluded[i]) {
 			memtx_tx_handle_gap_write(space, add_story, succ, i);
 			memtx_tx_handle_point_hole_write(space, add_story, i);
+		}
+		/* Form chains. */
+		if (next == NULL) {
```

Comment:
Why should we still link the story to this index?

Reply:
I mean it would be sufficient to set newer_story and older_story to NULL, but in this case `memtx_tx_story_link_top` does what needed.

Reply:
Discussed online, decided that since that link is already has NULLs (see `story_new`) there's no need to call link_top at all.